NodeProg/Infrastructure/NumberNode.cs

```diff
@@ -21,14 +22,129 @@ public NumberNode(long value)
                 .Min() ?? input.Length;
 
             var numberLength = stopMarkerIndex - index;
-            var numberString = input.Substring(index, numberLength);
 
-            long value;
-
-            if (Int64.TryParse(numberString.Trim(), out value))
-                return new NodeParseResult { Node = new NumberNode(value), StartPosition = index, Length = numberLength };
+            var value = TryParseInt64(input, index, numberLength);
+            if (value != null)
+                return new NodeParseResult { Node = new NumberNode(value.Value), StartPosition = index, Length = numberLength };
             return null;
+        }
+        private static long? TryParseInt64(
+            string input,
+            int startIndex,
+            int length,
+            bool allowLeadingWhitespace = true,
+            bool allowTrailingWhitespace = true)
+        {
+            if (length == 0)
+                return null;
+            var trimChars = new List<char>();
+            if (allowLeadingWhitespace)
```

Comment:
`allowLeadingWhitespace` means that white space characters are allowed in the beginning of the string. See parameters in #3.